velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -639,4 +639,74 @@ struct ArrayFlattenFunction {
     }
   }
 };
+
+/// This class implements the array remove function.
+///
+/// DEFINITION:
+/// array_remove(x, element) → array
+/// Remove all elements that equal element from array x.
+template <typename T>
+struct ArrayRemoveFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T)
+
+  // Fast path for primitives.
+  template <typename Out, typename In, typename E>
+  void call(Out& out, const In& inputArray, E element) {
+    for (const auto& item : inputArray) {
+      if (element != item.value()) {
```

Comment:
what if item is null?
it.value() is not safe to always be called, you should first check item.has_value();